base/array.jl

```diff
@@ -863,22 +863,9 @@ function indexin(a::AbstractArray, b::AbstractArray)
     [get(bdict, i, 0) for i in a]
 end
 
-# findin (the index of intersection)
-function findin(a, b::UnitRange)
-    ind = Array(Int, 0)
-    f = first(b)
-    l = last(b)
-    for i = 1:length(a)
-        if f <= a[i] <= l
-            push!(ind, i)
-        end
-    end
-    ind
-end
-
 function findin(a, b)
     ind = Array(Int, 0)
-    bset = Set(b)
+    bset = unique(b)
     @inbounds for i = 1:length(a)
         a[i] in bset && push!(ind, i)
```

Comment:
Yeah you are right its `O(1)`. Somehow I was thinking it was a tree based rather than hash based lookup. I have been a bit busy trying to reason about the best way to add `unique` methods, which has led me into the territory of adding optimized `Set` methods as well (which might give us the best of both worlds in the implied tradeoff), but I like everything you wrote above.